### PR TITLE
Planning scene nullptr check

### PR DESCRIPTION
**Pull request by rafaelrojasmiliani**
### Description

Issue description [here](https://github.com/ros-planning/moveit/issues/2405).

- Now the code in `planning_scene_monitor.cpp`, `PlanningSceneMonitor::initialize` is easier to read.
- Now `CollisionPluginLoader::setupScene` shows a message when its `PlanningScenePtr` argument is `nullptr`.

### Checklist
- [   ] `scene_const_` is assigned just once.
- [  ] `collision_loader_.setupScene` is called once
- [  ] Test the code.

### CODE REVIEW

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp
Comment (felixvd) [inline moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:221]:
```suggestion
    // scene_const_ is set regardless if scene_ is null or not
```

File: moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp
Comment (felixvd) [inline moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp:225]:
```suggestion
      // The scene_ is loaded on the collision loader only if it was correctly instantiated
```

**Review by felixvd — COMMENTED** [review]
Just pointing out some typos. Please title your PR like a commit message, since that's how it will appear in the commit history. You'll find more info on how to write a good commit message [here](https://chris.beams.io/posts/git-commit/).

File: moveit_ros/planning/collision_plugin_loader/src/collision_plugin_loader.cpp
Comment (AndyZe) [inline moveit_ros/planning/collision_plugin_loader/src/collision_plugin_loader.cpp:112]: Sorry to nitpick this nice PR but it will help keep things tidy in the future.

Can you please change this to:

`static const std::string LOGNAME = "collision_detection";`   <--- Goes near top of file, around L38

`ROS_WARN_NAMED(LOGNAME, "Cannot setup scene, PlanningScenePtr is null.");`

**Review by AndyZe — COMMENTED** [review]
@rafaelrojasmiliani  please make these minor changes then let's get this merged!

File: moveit_ros/planning/collision_plugin_loader/src/collision_plugin_loader.cpp
Comment (felixvd) [reply moveit_ros/planning/collision_plugin_loader/src/collision_plugin_loader.cpp:112]: We have other files that don't set LOGNAME, so I'd be fine either way

**Comment by v4hn** [comment]
> We have other files that don't set LOGNAME, so I'd be fine either way

We do, but the consensus was (and still is afaik) to add LOGNAME whenever you touch logging statements. Instead of demanding this from contributors, it would be good if reviewing maintainers directly push these changes to the pull-request. ;)

**Review by AndyZe — APPROVED** [review]

**Comment by rafaelrojasmiliani** [comment]
should I do something else?

**Comment by v4hn** [comment]
Sorry @rafaelrojasmiliani , CI for the melodic-devel branch seems to be broken at the moment (https://github.com/ros-planning/moveit/pull/2567#issuecomment-809248712) and we are in the middle of migrating from Travis as well.

If you want this merged right away, please rebase to the `master` branch and we can merge it there (and possibly backport later).
Otherwise we will merge it here once CI is happy again.

**Comment by AndyZe** [comment]
I tried rebasing this but there is a massive diff between it and moveit `master`. Not sure what's going on. I expected it to be a very easy rebase.

**Comment by AndyZe** [comment]
Closing and re-opening to retrigger CI.

**Comment by v4hn** [comment]
CI was broken in `melodic-devel` for quite a while, that's why it failed here.
I just rebased the patch on top of `master` and proposed #2632, replacing this PR.
Once it's merged there, it can be backported to `melodic-devel` (possibly still as part of #2630).

**Pull request closed without merging**
